From: Takashi Iwai <tiwai@suse.de>
Subject: kABI workaround for HD-audio
Patch-mainline: Never, kABI workaround
References: git-fixes

The fix patch
  patches.suse/ALSA-hda-fix-jack-detection-with-Realtek-codecs-when.patch
introduced a new bit field in the exported struct hda_codec, and it
breaks kABI.  As a standard idiom, move it to the tail and protect
with __GENKSYMS__.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 include/sound/hda_codec.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/include/sound/hda_codec.h
+++ b/include/sound/hda_codec.h
@@ -253,8 +253,10 @@ struct hda_codec {
 	unsigned int force_pin_prefix:1; /* Add location prefix */
 	unsigned int link_down_at_suspend:1; /* link down at runtime suspend */
 	unsigned int relaxed_resume:1;	/* don't resume forcibly for jack */
-	unsigned int forced_resume:1; /* forced resume for jack */
 	unsigned int mst_no_extra_pcms:1; /* no backup PCMs for DP-MST */
+#ifndef __GENKSYMS__
+	unsigned int forced_resume:1; /* forced resume for jack */
+#endif
 
 #ifdef CONFIG_PM
 	unsigned long power_on_acct;
